
<!-- README.md is generated from README.Rmd. Please edit that file -->

# Liver disease atlas

<!-- badges: start -->

**Publication:**
[![DOI](https://img.shields.io/badge/DOI-10.1002/hep4.1797-red.svg)](https://doi.org/10.1002/hep4.1797)
**Data:**
[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.4407207.svg)](https://doi.org/10.5281/zenodo.4407207)
<!-- badges: end -->

This repository contains the code for the analyses presented in
*“[Transcriptomic cross-species analysis of chronic liver disease
reveals a consistent regulation pattern between humans and
mice](https://doi.org/10.1002/hep4.1797)”*.

Reproducibility of all analyses is ensured by the R package
[workflowr](https://github.com/jdblischak/workflowr) by deploying all
our analysis scripts at a dedicated
[webpage](https://saezlab.github.io/liver-disease-atlas/).

## Transcriptomic cross-species analysis of chronic liver disease reveals consistent regulation between humans and mice

### Abstract

Mouse models are frequently used to study chronic liver diseases (CLDs).
To assess their translational relevance, we quantified the similarity of
commonly used mouse models to human CLDs based on transcriptome data.
Gene‐expression data from 372 patients were compared with data from
acute and chronic mouse models consisting of 227 mice, and additionally
to nine published gene sets of chronic mouse models. Genes consistently
altered in humans and mice were mapped to liver cell types based on
single‐cell RNA‐sequencing data and validated by immunostaining.
Considering the top differentially expressed genes, the similarity
between humans and mice varied among the mouse models and depended on
the period of damage induction. The highest recall (0.4) and precision
(0.33) were observed for the model with 12‐months damage induction by
CCl4 and by a Western diet, respectively. Genes consistently
up‐regulated between the chronic CCl4 model and human CLDs were enriched
in inflammatory and developmental processes, and mostly mapped to
cholangiocytes, macrophages, and endothelial and mesenchymal cells.
Down‐regulated genes were enriched in metabolic processes and mapped to
hepatocytes. Immunostaining confirmed the regulation of selected genes
and their cell type specificity. Genes that were up‐regulated in both
acute and chronic models showed higher recall and precision with respect
to human CLDs than exclusively acute or chronic genes. Conclusion:
Similarly regulated genes in human and mouse CLDs were identified.
Despite major interspecies differences, mouse models detected 40% of the
genes significantly altered in human CLD. The translational relevance of
individual genes can be assessed at
<https://saezlab.shinyapps.io/liverdiseaseatlas/>.

------------------------------------------------------------------------

### How to cite?

> Holland CH, Ramirez Flores RO, Myllys M, Hassan R, Edlund K, Hofmann
> U, Marchan R, Cadenas C, Reinders J, Hoehme S, Seddek A, Dooley S,
> Keitel V, Godoy P, Begher-Tibbe B, Trautwein C, Rupp C, Mueller S,
> Longerich T, Hengstler JG<sup>\#</sup>, Saez-Rodriguez J<sup>\#</sup>,
> Ghallab A<sup>\#</sup>. “Transcriptomic cross-species analysis of
> chronic liver disease reveals consistent regulation between humans and
> mice.” *Hepatology Communications*. 2021. DOI:
> [10.1002/hep4.1797](https://doi.org/10.1002/hep4.1797).

<sup>\#</sup>*Shared senior authorship*
