---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

# Liver disease atlas

<!-- badges: start -->
<!-- **Publication:** [![DOI](https://img.shields.io/badge/DOI-10.1186%2Fs13073--019--0615--0-blue.svg)][pub] -->
**Data:** [![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.4407207.svg)](https://doi.org/10.5281/zenodo.4407207)
<!-- badges: end -->

This repository contains the code for the analyses presented in _"Transcriptomic cross-species analysis of chronic liver disease reveals a consistent regulation pattern between humans and mice"_.

Rendered Rmarkdown files of all analyses can be found [here](https://saezlab.github.io/liver-disease-atlas/index.html).

## Transcriptomic cross-species analysis of chronic liver disease reveals a consistent regulation pattern between humans and mice

### Abstract

#### Background and aims 
Mouse models are frequently used to study chronic liver diseases (CLD). To assess their translational relevance, we quantified the similarity of commonly used mouse models to human CLD based on transcriptome data.

#### Methods 
Gene expression data from 372 patients were compared to data from acute and chronic mouse models comprising 227 mice, and additionally to nine published gene sets of chronic mouse models. Genes consistently altered in humans and mice were mapped to liver cell types based on single-cell RNA-sequencing data and validated by immunostaining. 

#### Results 
Considering the top differentially expressed genes, the similarity between humans and mice varied among the mouse models and depended on the period of damage induction. The highest recall (0.4) and precision (0.33) were observed for the model with 12 months damage induction by CCl4 and by a Western type damage, respectively. Genes consistently upregulated between the chronic CCl4 model and human CLD were enriched in inflammatory and developmental processes, and mapped to cholangiocytes, macrophages, endothelial and mesenchymal cells. Downregulated genes were enriched in metabolic processes and mapped to hepatocytes. Immunostaining confirmed the regulation of selected genes and their cell type specificity. Genes that were upregulated in both acute and chronic models showed higher recall and precision with respect to human CLD than exclusively acute or chronic genes.

#### Conclusion 
Similarly regulated genes in human and mouse CLD were identified. Despite major interspecies differences, mouse models detected 40% of the genes significantly altered in human CLD. The translational relevance of individual genes can be assessed at https://saezlab.shinyapps.io/liverdiseaseatlas/.

-------

### How to cite?
>Holland CH, Ramirez Flores RO, Myllys M, Hassan R, Edlund K, Hofmann U, Marchan R, Cadenas C, Reinders J, Hoehme S, Seddek A, Dooley S, Keitel V, Godoy P, Begher-Tibbe B, Trautwein C, Rupp C, Mueller S, Longerich T, Hengstler JG^#^, Saez-Rodriguez J^#^, Ghallab A^#^. "Transcriptomic cross-species analysis of chronic liver disease reveals a consistent regulation pattern between humans and mice." _In preparation_. 2021.

^#^_Shared senior authorship_
