---
title: "Chronic CCl4+WTD mouse model"
output: 
  workflowr::wflow_html:
    code_folding: hide
editor_options:
  chunk_output_type: console
---

```{r chunk-setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  autodep = TRUE,
  cache = TRUE
)
```

```{r wall-time-start, cache=FALSE, include=FALSE}
# Track time spent on performing this analysis
start_time <- Sys.time()
```

## Introduction
Here we analysis a mouse model of CCl4 in combination with western type diet induced chronic liver disease.

## Libraries and sources
These libraries and sources are used for this analysis.
```{r libs-and-src, message=FALSE, warning=FALSE, cache=FALSE}
library(tidyverse)
library(tidylog)
library(here)

library(edgeR)
library(biobroom)
library(progeny)
library(dorothea)

library(janitor)
library(msigdf) # remotes::install_github("ToledoEM/msigdf@v7.1")

library(AachenColorPalette)
library(cowplot)
library(lemon)
library(patchwork)
library(VennDiagram)
library(gridExtra)
library(ggpubr)

options("tidylog.display" = list(print))
source(here("code/utils-rnaseq.R"))
source(here("code/utils-utils.R"))
source(here("code/utils-plots.R"))
```

Definition of global variables that are used throughout this analysis.
```{r analysis-specific-params, cache=FALSE}
# i/o
data_path <- "data/mouse-chronic-ccl4-wtd"
output_path <- "output/mouse-chronic-ccl4-wtd"

# graphical parameters
# fontsize
fz <- 9
```

## Preliminary exploratory analysis
### Library size
Barplot of the library size (total counts) for each of the samples.
```{r lib-size}
count_matrix <- readRDS(here(data_path, "count_matrix.rds"))

plot_libsize(count_matrix) +
  my_theme(fsize = fz)
```

### Count distribution
Violin plots of the raw read counts for each of the samples.
```{r "count-distribution"}
count_matrix <- readRDS(here(data_path, "count_matrix.rds"))
meta <- readRDS(here(data_path, "meta_data.rds"))

count_matrix %>%
  tdy("gene", "sample", "count", meta) %>%
  arrange(treatment) %>%
  ggplot(aes(
    x = fct_reorder(sample, as.numeric(treatment)), y = log10(count + 1),
    group = sample, fill = interaction(time, diet, treatment)
  )) +
  geom_violin() +
  theme(
    axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1),
    legend.position = "top"
  ) +
  labs(x = NULL) +
  my_theme(grid = "no", fsize = fz)
```

### PCA of raw data
PCA plot of raw read counts contextualized based on the time point and treatment. Before gene with a constant expression across all samples are removed and count values are transformed to log2 scale. Only the top 1000 most variable genes are used as features.
```{r pca-raw-data}
count_matrix <- readRDS(here(data_path, "count_matrix.rds"))
meta <- readRDS(here(data_path, "meta_data.rds"))

stopifnot(colnames(count_matrix) == meta$sample)

# remove constant expressed genes and transform to log2 scale
preprocessed_count_matrix <- preprocess_count_matrix(count_matrix)


pca_result <- do_pca(preprocessed_count_matrix, meta, top_n_var_genes = 1000)

plot_pca(pca_result, feature = "time") +
  plot_pca(pca_result, feature = "diet") +
  plot_pca(pca_result, feature = "treatment") &
  my_theme(fsize = fz)
```

## Data processing
### Normalization
Raw read counts are normalized by first filtering out lowly expressed genes, TMM normalization and finally logCPM transformation.
```{r normalization}
count_matrix <- readRDS(here(data_path, "count_matrix.rds"))
meta <- readRDS(here(data_path, "meta_data.rds"))

stopifnot(meta$sample == colnames(count_matrix))

dge_obj <- DGEList(count_matrix, group = meta$group)

# filter low read counts, TMM normalization and logCPM transformation
norm <- voom_normalization(dge_obj)

saveRDS(norm, here(output_path, "normalized_expression.rds"))
```

### PCA of normalized data
PCA plot of normalized expression data contextualized based on the time point and treatment. Only the top 1000 most variable genes are used as features.
```{r pca-norm-data}
expr <- readRDS(here(output_path, "normalized_expression.rds"))
meta <- readRDS(here(data_path, "meta_data.rds"))

pca_result <- do_pca(expr, meta, top_n_var_genes = 1000)

saveRDS(pca_result, here(output_path, "pca_result.rds"))

plot_pca(pca_result, feature = "time") +
  plot_pca(pca_result, feature = "diet") +
  plot_pca(pca_result, feature = "treatment") &
  my_theme(fsize = fz)
```

## Differential gene expression analysis
Since there are no replicates we cannot perform standard differential gene 
expression analysis. Instead we compute manually the logFC.
```{r running-limma}
# load expression and meta data
expr <- readRDS(here(output_path, "normalized_expression.rds"))
meta <- readRDS(here(data_path, "meta_data.rds"))

stopifnot(colnames(expr) == meta$sample)

deg_df <- expr %>%
  data.frame() %>%
  rownames_to_column("gene") %>%
  as_tibble() %>%
  transmute(
    gene,
    ccl_effect_12w = GSM2630026 - GSM2630025,
    wd_effect_12w = GSM2630027 - GSM2630025,
    combined_effect_12w = GSM2630028 - GSM2630025,
    ccl_effect_24w = GSM2630030 - GSM2630029,
    wd_effect_24w = GSM2630031 - GSM2630029,
    combined_effect_24w = GSM2630032 - GSM2630029,
  ) %>%
  pivot_longer(-gene, names_to = "contrast", values_to = "logFC") %>%
  mutate(contrast_reference = "combined_effect")

saveRDS(deg_df, here(output_path, "limma_result.rds"))
```

### Distribution of logFC
```{r volcano-plots}
df <- readRDS(here(output_path, "limma_result.rds"))

df %>%
  ggplot(aes(x = contrast, y = logFC)) +
  geom_violin() +
  my_theme(grid = "y", fsize = fz)
```

## Translation to HGNC symbols
For later comparisons to human data the mouse gene symbols are mapped to their human orthologs.
```{r translate-to-hgnc-symbols}
df <- readRDS(here(output_path, "limma_result.rds"))

mapped_df <- df %>%
  translate_gene_ids(from = "symbol_mgi", to = "symbol_hgnc") %>%
  drop_na() %>%
  # for duplicated genes, keep the one with the highest absolute logFC
  group_by(contrast_reference, contrast, gene) %>%
  slice_max(order_by = abs(logFC), n = 1, with_ties = F) %>%
  ungroup()

saveRDS(mapped_df, here(output_path, "limma_result_hs.rds"))
```

```{r wall-time-end, cache=FALSE, include=FALSE}
duration <- abs(as.numeric(difftime(Sys.time(), start_time, units = "secs")))
t <- print(sprintf("%02d:%02d", duration %% 3600 %/% 60, duration %% 60 %/% 1))
```
Time spend to execute this analysis: `r t` minutes.
