---
title: "Home"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---

# Transcriptomic cross-species analysis of chronic liver disease reveals consistent regulation between humans and mice

The results and the analysis scripts presented on this website ensures the reproducibility of all bioinformatics related findings presented in [_"Transcriptomic cross-species analysis of chronic liver disease reveals consistent regulation between humans and mice"_](https://doi.org/10.1002/hep4.1797). Please see below to get more information about the individual analysis and instructions on how to reproduce the results.

Abstract
--------

Mouse models are frequently used to study chronic liver diseases (CLDs). To assess their translational relevance, we quantified the similarity of commonly used mouse models to human CLDs based on transcriptome data. Gene‐expression data from 372 patients were compared with data from acute and chronic mouse models consisting of 227 mice, and additionally to nine published gene sets of chronic mouse models. Genes consistently altered in humans and mice were mapped to liver cell types based on single‐cell RNA‐sequencing data and validated by immunostaining. Considering the top differentially expressed genes, the similarity between humans and mice varied among the mouse models and depended on the period of damage induction. The highest recall (0.4) and precision (0.33) were observed for the model with 12‐months damage induction by CCl4 and by a Western diet, respectively. Genes consistently up‐regulated between the chronic CCl4 model and human CLDs were enriched in inflammatory and developmental processes, and mostly mapped to cholangiocytes, macrophages, and endothelial and mesenchymal cells. Down‐regulated genes were enriched in metabolic processes and mapped to hepatocytes. Immunostaining confirmed the regulation of selected genes and their cell type specificity. Genes that were up‐regulated in both acute and chronic models showed higher recall and precision with respect to human CLDs than exclusively acute or chronic genes. Conclusion: Similarly, regulated genes in human and mouse CLDs were identified. Despite major interspecies differences, mouse models detected 40% of the genes significantly altered in human CLD. The translational relevance of individual genes can be assessed at https://saezlab.shinyapps.io/liverdiseaseatlas/.


Analysis
--------

The tab `Mouse models` contains Rmarkdown scripts to analyze and characterize the transcriptomic profiles of acute and chronic liver disease mouse models. These analyses comprised

* Normalization
* PCA analysis
* Differential gene expression analysis
* Time series clustering and characterization (if applicable)

The tab `Patient cohorts` contains Rmarkdown scripts to analyze and characterize the transcriptomic profiles of patient cohorts suffering from various chronic liver disease etiologies. These analyses comprised:

* Normalization
* PCA analysis
* Differential gene expression analysis

The tab `Meta analysis` contains Rmarkdown scripts to integrate acute and chronic mouse models with patient cohorts.

* [Chronic vs. acute](meta-chronic-vs-acute.html)
  * Identification of exclusively and commonly- regulated genes of chronic and acute disease in mice.
  
* [Mouse vs. human](meta-mouse-vs-human.html)
  * Identification of consistently regulated genes in the chronic CCl~4~[mouse-chronic-ccl4.html] mouse model and patients.
  * Quantification of the similarity of the in total 12 chronic mouse models with the different human patient cohorts based on precision and recall.
  
The tab `Figures` contains Rmarkdown scripts to generate the figures used in the manuscript.


How can I reproduce the analyses?
--------
We have used the [workflowr](https://jdblischak.github.io/workflowr/) package to organize the analysis scripts within this project so please familiarize yourself with its concept.

1. Clone the repository from https://github.com/saezlab/liver-disease-atlas which automatically provides you with all analysis scripts and the directory structure.
2. You need to install all packages that are required for the analyses. The package [renv](https://rstudio.github.io/renv/articles/renv.html) allows you to easily install the packages with the correct versions:

```{r eval=FALSE}
install.packages("renv")
renv::restore()
```

3. The GitHub repository contains only the analysis code and same small objects. All raw data is deposited at Zenodo [![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.4407207.svg)](https://doi.org/10.5281/zenodo.4407207).
Download the zipped `data` folder unzip it and replace the existing `data` folder at the root level of the R-project.

4. Run all analyses by running
```{r eval=FALSE}
install.packages("workflowr")
workflowr::wflow_build(republish = TRUE)
```
All intermediate and final results will be saved in the `output` folder. Please make sure to run _all_ analyses in the specified order as downstream analyses depend on previously generated results.

How to cite?
--------
>Holland CH, Ramirez Flores RO, Myllys M, Hassan R, Edlund K, Hofmann U, Marchan R, Cadenas C, Reinders J, Hoehme S, Seddek A, Dooley S, Keitel V, Godoy P, Begher-Tibbe B, Trautwein C, Rupp C, Mueller S, Longerich T, Hengstler JG^#^, Saez-Rodriguez J^#^, Ghallab A^#^. "Transcriptomic cross-species analysis of chronic liver disease reveals consistent regulation between humans and mice." _Hepatology Communications_. 2021. DOI: [10.1002/hep4.1797](https://doi.org/10.1002/hep4.1797).

^#^_Shared senior authorship_
